File: Searching-Algorithms/BinarySearch/gelvis/gelvis.js
let binarySearch = (array,value) =>{
    let upperBound = array.length - 1,lowerBound = 0;
    while (lowerBound <= upperBound){
        let middlePoint = (lowerBound + upperBound)/2;
        if(value < array[middlePoint])
            upperBound = middlePoint - 1;
        else if(value > array[middlePoint])
            lowerBound = middlePoint + 1;
        else
            return middlePoint;
    }
    return -1;
}

File: Searching-Algorithms/BinarySearch/gelvis/gelvis.cpp
#include <iostream>
// implementation for binary search
int binarySearch(int arr[], int searchValue, int arrLength)
{
    int lower_bound = 0, upper_bound = arrLength - 1;

    while (lower_bound <= upper_bound)
    {
        int middleIndex = (lower_bound + upper_bound) / 2;
        if (arr[middleIndex] == searchValue)
            return middleIndex;
        else if (arr[middleIndex] < searchValue)
            lower_bound = middleIndex + 1;
        else
            upper_bound = middleIndex - 1;
    }
    return -1;
}

int main()
{
    int arr[] = {1, 2, 3, 4, 5};
    std::cout << binarySearch(arr, 2, 5) << std::endl;
}